Schemata/sarif-schema.json

```diff
@@ -353,6 +272,54 @@
       }
     },
 
+    "externalFiles": {
+      "description": "References to external files that should be inlined with the content of a root log file.",
+      "additionalProperties": false,
+      "type": "object",
+      "properties": {
+
+        "conversion": {
+          "description": "An external reference to a run.conversion object to be merged with the root log file.",
+          "$ref": "#/definitions/fileLocation"
+        },
+
+        "files": {
+          "description": "An external reference to a run.files object to be merged with the root log file.",
+          "$ref": "#/definitions/fileLocation"
+        },
+
+        "graphs": {
+          "description": "An external reference to a run.graphs object to be merged with the root log file.",
+          "$ref": "#/definitions/fileLocation"
+        },
+
+        "invocations": {
+          "description": "An external reference to an run.array of invocation objects to be merged with the root log file.",
+          "$ref": "#/definitions/fileLocation"
```

Comment:
>"$ref": "#/definitions/fileLocation" [](start = 10, length = 36)

... and since the `run.invocations` property _is_ an array, its value must be an array of `fileLocation` objects, just as for `run.results`. I hadn't noticed this before either; I thought `results` was the only array-valued externalizable property.

Reply:
right!

---
In reply to: [215686784](https://github.com/oasis-tcs/sarif-spec/pull/236#discussion_r215686784) [](ancestors = 215686784)

Reply:
This one isn't resolved; it still specifies a single `fileLocation` instead of an array of `fileLocation`s. Everything else looks good, and I agree with "Pending" on the "graphs" problem.

---
In reply to: [215779432](https://github.com/oasis-tcs/sarif-spec/pull/236#discussion_r215779432) [](ancestors = 215779432,215686784)

Reply:
As we discussed:
1. An `invocation` might be large because it contains `toolNotification`.
2. Each of the external files for any array-valued property contains an _array_, and it's the union of those arrays that is merged with the root file. So if all the objects are small, you can put them in one external file.

---
In reply to: [215783343](https://github.com/oasis-tcs/sarif-spec/pull/236#discussion_r215783343) [](ancestors = 215783343,215780861,215779432,215686784)